lib/js/components/Buttons/IconButton/IconButton.vue

```diff
@@ -50,42 +57,83 @@
 @import '../../../../styles/settings/spacings';
 @import '../../../../styles/settings/typography';
 
-@mixin iconButtonColor($color, $hover-color: '') {
-	@if $hover-color == '' {
-		//$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
-	}
-
-	color: $color;
-
-	&:hover,
-	&:active {
-		color: $hover-color;
-	}
-}
-
 .a-iconButton {
 	$self: &;
 
-	@each $color-name, $color-map in $theme-calculated-colors {
+	@each $color-name, $color-map in $icon-button-colors {
 		&.-color-#{$color-name} {
-			@include iconButtonColor(map-get($color-map, 'color'), map-get($color-map, 'hovered'));
-		}
-	}
+			#{$self}__label {
+				color: map-get($color-map, 'outlined', 'color');
+			}
 
-	@each $color-name, $color in $regular-colors {
-		&.-color-#{$color-name} {
-			@include iconButtonColor($color);
+			&:hover,
+			&.-hovered {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-hovered');
+					color: map-get($color-map, 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-hovered');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-hovered');
+				}
+			}
+
+			&:focus,
+			&.-focused {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-focused');
+					color: map-get($color-map, 'filled', 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-focused');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-focused');
+				}
+			}
+
+			&.-disabled {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-disabled');
+					color: map-get($color-map, 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background');
```

Comment:
do we need to override original value here?

Reply:
Yes, because background in button has color (same color as here icon and border), we are set here ghost background (white).